content/ui/popup.js

```diff
@@ -0,0 +1,197 @@
+
+  setSessionTimer();
+  resizeDropdown();
+
+  registerForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+    register();
+  });
+
+  loginForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+    login();
+  });
+
+  logoutForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+    logout();
+  });
+});
+
+
+//register
+const register = () => {
+  const cpasswordInput = document.getElementById("cpassword");
+  const passwordInput = document.getElementById("password");
+  const statusDiv = document.getElementById("status");
+
+  const pass1 = cpasswordInput.value;
+  const pass2 = passwordInput.value;
+
+  statusDiv.textContent = "";
+
+  chrome.runtime.sendMessage(
+    { action: "register", cpassword: pass1, password: pass2 },
+    (response) => {
+      if (response && response.success === true) {
+        statusDiv.textContent = "Password set successfully!";
+        updatePopupContent();
+      } else {
+        statusDiv.textContent = response.error;
+        cpasswordInput.value = "";
+        passwordInput.value = "";
+      }
+    }
+  );
+};
+
+//login
+const login = async () => {
+  const sessionTimeSelect = document.getElementById("sessionTime");
+  const loginPassword = document.getElementById("loginPassword");
+  const statusDiv = document.getElementById("status");
+  const passwordInput = document.getElementById("password");
+  const socialToggle = document.getElementById("blockSocialMediaCheckbox");
+  const gamingToggle = document.getElementById("blockGamesCheckbox");
+
+  //array to store toggle elements and their checked status for service worker injection
+  const toggles = [
+    { id: "socialMediaToggle", element: socialToggle, checked: false },
+    { id: "gamingToggle", element: gamingToggle, checked: false },
+  ];
+  toggles.forEach((toggle) => {
+    toggle.checked = toggle.element.checked;
+  });
+  const checkedToggles = toggles.filter((toggle) => toggle.checked);
+
+  const pass = loginPassword.value;
+  statusDiv.textContent = "";
+
+  const sessionTime = sessionTimeSelect.value;
+  // Send login request to background script
+  await chrome.runtime.sendMessage(
+    { action: "login", password: pass, checkedToggles: checkedToggles, sessionTime: sessionTime },
+    (response) => {
+      if (response && response.success === true) {
+        statusDiv.textContent = "Logged in successfully!";
+        updatePopupContent();
+
+      } else {
+        statusDiv.textContent = response.error;
+        passwordInput.value = "";
+      }
+    }
+  );
+};
+
+//logout
+const logout = () => {
+  const logoutPassword = document.getElementById("logoutPassword");
+  const statusDiv = document.getElementById("status");
+  const passwordInput = document.getElementById("password");
+
+  const pass2 = logoutPassword.value;
+  statusDiv.textContent = "";
+
+  chrome.runtime.sendMessage(
+    { action: "logout", password: pass2 },
+    (response) => {
```

Comment:
This callback can be made reusable since the handling for response is very similar in this file. we can create a new function which handles the response and updates content